types/ember/index.d.ts

```diff
@@ -3281,6 +3281,33 @@ declare module 'ember' {
         /**
          * Returns a consistent type for the passed object.
          */
+        function typeOf(item: string): 'string';
+        function typeOf(item: number): 'number';
+        function typeOf(item: boolean): 'boolean';
+        function typeOf(item: null): 'null';
+        function typeOf(item: undefined): 'undefined';
+        function typeOf(item: (...args: any[]) => any): 'function';
+        function typeOf(item: any[]): 'array';
+        function typeOf(item: RegExp): 'regexp';
+        function typeOf(item: Date): 'date';
+        function typeOf(item: Error): 'error';
+        function typeOf<T extends object>(item: T): T extends FileList
```

Comment:
Is there an advantage to write it this way instead of
```ts
function typeOf(item: FileList): 'filelist';
function typeOf(item: typeof Ember.Object): 'class';
function typeOf(item: Number): 'number';
// etc
```
?

Reply:
Unfortunately, with the strategy we both prefer, a lot of things were falling into the 'object' case, despite having more specific function heads available.

Reply:
Even when the 'object' case is listed after the others?

Reply:
I fiddled with this some more and got a function overload order that works. Thanks for poking me to get this done the cleaner way!